utils/tracing/go/tracing.go

```diff
@@ -1,104 +1,116 @@
-	tp := sdktrace.NewTracerProvider(
-		sdktrace.WithSampler(sdktrace.AlwaysSample()),
-		sdktrace.WithSyncer(exporter),
-		sdktrace.WithResource(resource.NewWithAttributes(
-			semconv.ServiceNameKey.String(fmt.Sprintf("%v@%v", serviceName, hostname)),
+	exporter, err := newZipkinExporter(url, logger)
+	tp := trace.NewTracerProvider(
+		trace.WithSampler(trace.AlwaysSample()),
+		trace.WithSyncer(exporter),
+		trace.WithResource(resource.NewWithAttributes(
+			semconv.ServiceNameKey.String(fmt.Sprintf("%v@%v-FU", serviceName, hostname)),
 			attribute.Int64("ID", 1),
 		)),
 	)
 	f := initTracer(tp)
-	return f
+	return f, err
 }
 
-// InitCustomTracer initialises an opentelemetry tracer. It uses a zipkin exporter which exports
-// to the given url and logger, and samples traces at the specified traceRate. Any number of additional
-// attributes (attr) can be provided.
-func InitCustomTracer(url string, traceRate float64, logger *log.Logger, attr ...attribute.KeyValue) func() {
-	exporter := newZipkinExporter(url, logger)
-	var sampler sdktrace.Sampler
+// InitCustomTracer initialises an OpenTelemetry tracer. It uses a zipkin exporter which exports
+// to the given URL and logger, and samples traces at the specified traceRate. Any number of additional
+// attributes (attr) can be provided. Note that this tracer uses the default name, which will be the name
+// of the process according to the OS, so to name your tracer provide a "service.name" attribute.
+func InitCustomTracer(url string, traceRate float64, logger *default_log.Logger, attr ...attribute.KeyValue) (func(), error) {
+	exporter, err := newZipkinExporter(url, logger)
+	var sampler trace.Sampler
 	if traceRate >= 1 {
-		sampler = sdktrace.AlwaysSample()
+		sampler = trace.AlwaysSample()
 	} else {
-		sampler = sdktrace.ParentBased(sdktrace.TraceIDRatioBased(traceRate))
+		sampler = trace.ParentBased(trace.TraceIDRatioBased(traceRate))
 	}
-	tp := sdktrace.NewTracerProvider(
-		sdktrace.WithSampler(sampler),
-		sdktrace.WithSyncer(exporter),
-		sdktrace.WithResource(resource.NewWithAttributes(
+	tp := trace.NewTracerProvider(
+		trace.WithSampler(sampler),
+		trace.WithSyncer(exporter),
+		trace.WithResource(resource.NewWithAttributes(
 			attr...,
 		)),
 	)
 	f := initTracer(tp)
-	return f
-}
-
-// Span gives a simple abstraction for making multiple opentelemetry spans within your process.
-type Span struct {
-	SpanName, TracerName string
-	traceSpan            *trace.Span
-}
-
-// StartSpan starts the span. ctx is used to determine parent spans, IDs, etc.
-// Returns a context which must be used in child spans such that they appear
-// in the hierarchy.
-func (s *Span) StartSpan(ctx context.Context) context.Context {
-	tracer := otel.GetTracerProvider().Tracer(s.TracerName)
-	newctx, traceSpan := tracer.Start(ctx, s.SpanName)
-	s.traceSpan = &traceSpan
-	return newctx
-}
-
-// EndSpan ends the span.
-func (s *Span) EndSpan() {
-	(*s.traceSpan).End()
+	return f, err
```

Comment:
```suggestion
	return f, nil
```
if the code got there, there was no error